src/nemos/regularizer.py

```diff
@@ -74,181 +45,68 @@ class Regularizer(Base, abc.ABC):
-            strength for proximal gradient methods.
-
-        prox:
-            Optional, the proximal projection operator.
-
-        *kwargs:
-            Keyword arguments for the jaxopt `solver.run` method.
+        Abstract method to retrieve the proximal operator for this solver.
 
         Returns
         -------
         :
-            A tuple containing three callable functions:
-            - solver_init_state: Function to initialize the solver's state, necessary before starting the optimization.
-            - solver_update: Function to perform a single update step in the optimization process,
-            returning new parameters and state.
-            - solver_run: Function to execute the optimization process, applying multiple updates until a
-            stopping criterion is met.
+            The proximal operator, which typically applies a form of regularization.
         """
-        # check that the loss is Callable
-        utils.assert_is_callable(loss, "loss")
-
-        # get the solver with given arguments.
-        # The "fun" argument is not always the first one, but it is always KEYWORD
-        # see jaxopt.EqualityConstrainedQP for example. The most general way is to pass it as keyword.
-        # The proximal gradient is added to the kwargs if passed. This avoids issues with over-writing
-        # the proximal operator.
-        if "prox" in self.solver_kwargs:
-            if prox is None:
-                raise ValueError(
-                    f"Regularizer of type {self.__class__.__name__} "
-                    f"does not require a proximal operator!"
-                )
-            else:
-                warnings.warn(
-                    "Overwritten the user-defined proximal operator! "
-                    "There is only one valid proximal operator for each regularizer type.",
-                    UserWarning,
-                )
-        # update the kwargs if prox is passed
-        if prox is not None:
-            solver_kwargs = self.solver_kwargs.copy()
-            solver_kwargs.update(prox=prox)
-        else:
-            solver_kwargs = self.solver_kwargs
-        solver = getattr(jaxopt, self.solver_name)(fun=loss, **solver_kwargs)
-
-        def solver_run(
-            init_params: Tuple[DESIGN_INPUT_TYPE, jnp.ndarray], *run_args: jnp.ndarray
-        ) -> jaxopt.OptStep:
-            return solver.run(init_params, *args, *run_args, **kwargs)
-
-        def solver_update(params, state, *run_args, **run_kwargs) -> jaxopt.OptStep:
-            return solver.update(
-                params, state, *args, *run_args, **kwargs, **run_kwargs
-            )
-
-        def solver_init_state(params, state, *run_args, **run_kwargs) -> NamedTuple:
-            return solver.init_state(
-                params, state, *args, *run_args, **kwargs, **run_kwargs
-            )
-
-        return solver_init_state, solver_update, solver_run
+        pass
 
 
 class UnRegularized(Regularizer):
```

Comment:
Unregularized has the identity as a proximal operator, this class as any other regularizer should have the get_proximal_operator method too